napari/_vispy/overlays/scale_bar.py

```diff
@@ -15,7 +15,6 @@ class VispyScaleBarOverlay(ViewerOverlayMixin, VispyCanvasOverlay):
     def __init__(self, *, viewer, overlay, parent=None) -> None:
         self._target_length = 150
         self._scale = 1
-        self._unit = None
```

Comment:
Nit: even though this is eventually initialized in `reset` that requires digging through code. I believe the goal here is to elevate the declaration/definition of all attributes in `__init__` to clarify their existence in one place, which is something I like. Alternatively,  I think we could make them part of the class definition above `__init__`?

Reply:
Agree on keeping this around here.

Reply:
Tried adding this back, but I maybe see why it was removed now. If we assign to `None` here, then this is an optional unit. If we let `reset` do it, then it's not optional which is consistent with its internal usage (i.e. we never check if it's None before accessing its attributes/methods).

I'd be fine with assigning to `self._unit = get_unit_registry()(overlay.unit)` though that is a little wasteful because we'll make the same call in `__init__`. What about just declaring the type with `self._unit: pint.Unit`?

Reply:
👍 I put the type hint in `__init__`